src/EventStateMachineFactory.php

```diff
@@ -142,43 +162,113 @@ class EventStateMachineFactory extends AbstractBaseCallbackFactory implements St
     const K_CFG_EVENT_PARAMS = 'event_params';
 
     /**
-     * {@inheritdoc}
+     * The event manager to use for created instances.
      *
      * @since [*next-version*]
+     *
+     * @var null|EventManagerInterface
      */
-    public function make($config = null)
-    {
-        return parent::make($config);
+    protected $eventManager;
+
+    /**
+     * The event factory to use for created instances.
+     *
+     * @since [*next-version*]
+     *
+     * @var null|EventFactoryInterface
+     */
+    protected $eventFactory;
+
+    /**
+     * The event name format to use for created instances.
+     *
+     * @since [*next-version*]
+     *
+     * @var Stringable|null|string
+     */
+    protected $eventNameFormat;
+
+    /**
+     * The event params to use for created instances.
+     *
+     * @since [*next-version*]
+     *
+     * @var array|null
+     */
+    protected $eventParams;
+
+    /**
+     * The event target to use for created instances.
+     *
+     * @since [*next-version*]
+     *
+     * @var mixed|null
+     */
+    protected $eventTarget;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param EventManagerInterface|null      $eventManager    The event manager to use for created instances.
+     * @param EventFactoryInterface|null      $eventFactory    The event factory to use for created instances.
+     * @param string|Stringable|null          $eventNameFormat The event name format to use for created instances.
+     * @param array|stdClass|Traversable|null $eventParams     The event params to use for created instances.
+     * @param mixed|null                      $eventTarget     The event target to use for created instances.
+     */
+    public function __construct(
+        EventManagerInterface $eventManager = null,
+        EventFactoryInterface $eventFactory = null,
+        $eventNameFormat = null,
+        $eventParams = null,
+        $eventTarget = null
+    ) {
+        $this->eventManager = $eventManager;
+        $this->eventFactory = $eventFactory;
+        $this->eventTarget  = $eventTarget;
+
+        $this->eventNameFormat = ($eventNameFormat !== null)
+            ? $this->_normalizeStringable($eventNameFormat)
+            : null;
+
+        $this->eventParams = ($eventParams !== null)
+            ? $this->_normalizeArray($eventParams)
```

Comment:
As I mention below, normalize to iterable (or container, whichever is appropriate). Similar to normalizing to stringable VS normalizing to string.

Reply:
Hmm. Given that events only accept arrays, this wouldn't make sense. Their interface promises array params. So perhaps a better solution would be to normalize the params in the state machine (when the event is created via the factory), or perhaps normalize in the event instance itself. Maybe the former is more appropriate because it doesn't assume that the event implementation will normalize.

Reply:
For reference, I mean [here](https://github.com/RebelCode/event-state-machine/blob/develop/src/EventStateMachine.php#L550).

Reply:
Ya, you can normalize to array there. Point is that in the constructor of this class, no need to limit to array.